bigframes/operations/_matplotlib/core.py

```diff
@@ -78,3 +87,55 @@ class ScatterPlot(SamplingPlot):
     @property
     def _kind(self) -> typing.Literal["scatter"]:
         return "scatter"
+
+    def __init__(self, data, **kwargs) -> None:
+        super().__init__(data, **kwargs)
+
+        c = self.kwargs.get("c", None)
+        if self._is_sequence_arg(c) and len(c) != self.data.shape[0]:
```

Comment:
Can we handle alignment at an earlier level and just accept `c` being a column id present in the data?

Reply:
As we discussed offline, we can ask users to add this list-like colors to the dataframe before plotting and pass the column name.